File: backend/app/schemas/mrp.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class MrpComputeRequest(BaseModel):
    work_order_ids: list[int]
    remark: str | None = None


class MrpItemOut(BaseModel):
    id: int
    work_order_id: int | None
    order_id: int | None
    sku_id: int
    material_id: int
    bom_id: int | None
    bom_scope: str | None
    wo_qty: int
    qty_per: int
    gross_qty: int
    stock_qty: int
    net_qty: int
    suggested_purchase_qty: int
    supplier_id: int | None
    unit_price: Decimal | None

    work_order_code: str | None = None
    order_code: str | None = None
    sku_code: str | None = None
    sku_name: str | None = None
    material_code: str | None = None
    material_name: str | None = None
    material_unit: str | None = None
    supplier_name: str | None = None


class MrpPlanOut(BaseModel):
    id: int
    code: str
    status: str
    source_type: str
    remark: str | None
    total_skus: int
    total_materials: int
    total_purchase_qty: int
    created_by: int | None
    created_at: datetime
    items: list[MrpItemOut] = []


class MrpPlanBrief(BaseModel):
    id: int
    code: str
    status: str
    source_type: str
    total_skus: int
    total_materials: int
    total_purchase_qty: int
    created_at: datetime

File: frontend-admin-pro/src/api/mrp.ts
import { http } from '@/utils/http'
import type { ListResp } from '@/types/api'

export interface MrpItemOut {
  id: number
  work_order_id: number | null
  order_id: number | null
  sku_id: number
  material_id: number
  bom_id: number | null
  bom_scope: string | null
  wo_qty: number
  qty_per: number
  gross_qty: number
  stock_qty: number
  net_qty: number
  suggested_purchase_qty: number
  supplier_id: number | null
  unit_price: string | null
  work_order_code: string | null
  order_code: string | null
  sku_code: string | null
  sku_name: string | null
  material_code: string | null
  material_name: string | null
  material_unit: string | null
  supplier_name: string | null
}

export interface MrpPlanBrief {
  id: number
  code: string
  status: string
  source_type: string
  total_skus: number
  total_materials: number
  total_purchase_qty: number
  created_at: string
}

export interface MrpPlanOut {
  id: number
  code: string
  status: string
  source_type: string
  remark: string | null
  total_skus: number
  total_materials: number
  total_purchase_qty: number
  created_at: string
  items: MrpItemOut[]
}

export interface MrpComputeIn {
  work_order_ids: number[]
  remark?: string | null
}

export const mrpApi = {
  listPlans(params: any) {
    return http.request<ListResp<MrpPlanBrief>>({ url: '/admin/mrp', method: 'GET', params })
  },
  getPlan(id: number) {
    return http.request<MrpPlanOut>({ url: `/admin/mrp/${id}`, method: 'GET' })
  },
  compute(data: MrpComputeIn) {
    return http.request<{ id: number; code: string }>({ url: '/admin/mrp/compute', method: 'POST', data })
  },
}
